pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -4302,6 +4319,702 @@ func Test_server_MultiInsert(t *testing.T) {
+			for _, r := range req.Requests {
+				uuids = append(uuids, r.Vector.Id)
+			}
+
+			return test{
+				name: "Boundary Value Testing case 4.1: Fail to MultiInsert with 1 request with empty UUID (vector type is uint8)",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:              name,
+					ip:                ip,
+					svcCfg:            defaultIntSvcCfg,
+					svcOpts:           defaultSvcOpts,
+					streamConcurrency: 0,
+				},
+				want: want{
+					err: status.WrapWithInvalidArgument(fmt.Sprintf("MultiInsert API invalid uuids \"%v\" detected", uuids), errors.ErrUUIDNotFound(0),
+						&errdetails.RequestInfo{
+							RequestId:   strings.Join(uuids, ", "),
+							ServingData: errdetails.Serialize(req),
+						},
+						&errdetails.BadRequest{
+							FieldViolations: []*errdetails.BadRequestFieldViolation{
+								{
+									Field:       "uuid",
+									Description: errors.ErrUUIDNotFound(0).Error(),
+								},
+							},
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: ngtResourceType + "/ngt.MultiInsert",
+							ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+						}),
+				},
+			}
+		}(),
+		func() test {
+			req := genF32Req(vector.Gaussian, 100, f32VecDim)
+			req.Requests[0].Vector.Id = ""
+
+			uuids := make([]string, 0, len(req.Requests))
+			for _, r := range req.Requests {
+				uuids = append(uuids, r.Vector.Id)
+			}
+
+			return test{
+				name: "Boundary Value Testing case 4.2: Fail to MultiInsert with 1 request with empty UUID (vector type is float32)",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:              name,
+					ip:                ip,
+					svcCfg:            defaultF32SvcCfg,
+					svcOpts:           defaultSvcOpts,
+					streamConcurrency: 0,
+				},
+				want: want{
+					err: status.WrapWithInvalidArgument(fmt.Sprintf("MultiInsert API invalid uuids \"%v\" detected", uuids), errors.ErrUUIDNotFound(0),
+						&errdetails.RequestInfo{
+							RequestId:   strings.Join(uuids, ", "),
+							ServingData: errdetails.Serialize(req),
+						},
+						&errdetails.BadRequest{
+							FieldViolations: []*errdetails.BadRequestFieldViolation{
+								{
+									Field:       "uuid",
+									Description: errors.ErrUUIDNotFound(0).Error(),
+								},
+							},
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: ngtResourceType + "/ngt.MultiInsert",
+							ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+						}),
+				},
+			}
+		}(),
+		func() test {
+			req := genIntReq(vector.Gaussian, 100, intVecDim)
+			for i := 0; i < 100; i += 2 {
+				req.Requests[i].Vector.Id = ""
+			}
+
+			uuids := make([]string, 0, len(req.Requests))
+			for _, r := range req.Requests {
+				uuids = append(uuids, r.Vector.Id)
+			}
+
+			return test{
+				name: "Boundary Value Testing case 4.3: Fail to MultiInsert with 50 request with empty UUID (vector type is uint8)",
```

Comment:
I forgot why a half vectors with no uuid.
let me know the reason.

Reply:
The reason I said was to ensure if part of the request will not be proceeded, and ensure if multiple errors occurred in one multiinsert request, how Vald handles it and no fatal error will be occurred